File: CSharpCodeBase/components/animations/mecanim.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local MecanimComponent = Class({__includes=Component, init = function (self, entity)
      Component.init(self, entity)
      self.animator = self.transform:GetComponent("Animator")
      self.entity.mecanim = self
end})

function MecanimComponent:ForceSetState(name, layer)
  self.animator:CrossFade(name, 0, layer or -1, 0)
end

function MecanimComponent:ForceSetNewState(name, time)
  self.animator:CrossFade(name, 0, -1, time)
end

function MecanimComponent:SetFloat(name, val)
  self.animator:SetFloat(name, val)
end

function MecanimComponent:SetBool(name, val)
  self.animator:SetBool(name, val)
end

function MecanimComponent:ResetTrigger(name)
  self.animator:ResetTrigger(name)
end

function MecanimComponent:SetTrigger(name)
  self.animator:SetTrigger(name)
end

function MecanimComponent:CheckStateName(name, layer)
  return MecanimUtils.CheckCurrentStateName(self.animator, name, layer or 0)
end

function MecanimComponent:GetFloat(name)
  return self.animator:GetFloat(name)
end

return MecanimComponent

File: CSharpCodeBase/components/animations/mecanim.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class mecanim :component{
 var MecanimComponent = Class({__includes=Component, init = public void (self, entity){
       Component.init(self, entity);
       this.animator = this.transform:GetComponent("Animator");
       this.entity.mecanim = self;
 }})
 public void MecanimComponent:ForceSetState(name, layer){
   this.animator:CrossFade(name, 0, layer  ||  -1, 0);
 }
 public void MecanimComponent:ForceSetNewState(name, time){
   this.animator:CrossFade(name, 0, -1, time);
 }
 public void MecanimComponent:SetFloat(name, val){
   this.animator:SetFloat(name, val);
 }
 public void MecanimComponent:SetBool(name, val){
   this.animator:SetBool(name, val);
 }
 public void MecanimComponent:ResetTrigger(name){
   this.animator:ResetTrigger(name);
 }
 public void MecanimComponent:SetTrigger(name){
   this.animator:SetTrigger(name);
 }
 public void MecanimComponent:CheckStateName(name, layer){
   Utils.CheckCurrentStateName(this.animator, name, layer  ||  0);
 }
 public void MecanimComponent:GetFloat(name){
   return this.animator:GetFloat(name);
 }
 Component}}